keras/preprocessing/image.py

```diff
@@ -822,22 +835,30 @@ def next(self):
             index_array, current_index, current_batch_size = next(self.index_generator)
         # The transformation of images is not under thread lock
         # so it can be done in parallel
-        batch_x = np.zeros((current_batch_size,) + self.image_shape)
+        # If image_shape has None in position 1 it is intended to be used as a FCN
+        if self.image_shape[1]:
+            batch_x = np.zeros((current_batch_size,) + self.image_shape)
         grayscale = self.color_mode == 'grayscale'
         # build batch of image data
         for i, j in enumerate(index_array):
             fname = self.filenames[j]
             img = load_img(os.path.join(self.directory, fname),
                            grayscale=grayscale,
                            target_size=self.target_size)
-            x = img_to_array(img, dim_ordering=self.dim_ordering)
+            x = img_to_array(img, dim_ordering=self.dim_ordering, color_mode=self.color_mode)
+            # Allows shape to be None, but batch size must equal one.
+            if not self.image_shape[1] and i == 0:
+                if current_batch_size != 1:
```

Comment:
Why must batch size be one? Just looking for clarification because I'm not 100% sure what this case is checking for because dense prediction tasks including FCNs can train with batch sizes >1.